lua/xcodebuild/tests/provider.lua

```diff
@@ -36,7 +36,7 @@ function M.find_tests(opts)
   local selectedTests = {}
 
   for _, line in ipairs(lines) do
-    selectedClass = string.match(line, "class ([^:%s]+)%s*:?")
+    selectedClass = string.match(line, "^[^\\/]*class ([^:%s]+)%s*:?")
```

Comment:
Actually, I just realized that there is a mistake here. You escape characters in lua pattern using `%` not `\`. So this pattern actually excludes both `\` and `/`. Could you fix that?

It should be `[^/]` or `[^%/]`. I'm not sure if `/` needs to be escaped.

Reply:
I checked and `/` doesn't have to be escaped. Only these characters:
```
( ) . % + - * ? [ ^ $
```

Reply:
Interesting! I didn't know that regex escaping varies between languages. Done.

Reply:
It doesn't, but it's not regex :D It's lua pattern, seems similar but it's not the same